Services/BoardFactory.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Chess_Backend.Models;
+using Chess_Backend.Models.Movements;
+using Chess_Backend.Models.Pieces;
+using Chess_Backend.Models.Positions;
+
+namespace Chess_Backend.Services
+{
+    public class BoardFactory : IBoardFactory
+    {
+
+        
+        private readonly IPieceFactory pieceFactory;
+        public BoardFactory(IPieceFactory pieceFactory)
+        {
+            this.pieceFactory = pieceFactory;
+        }
+
+
+        public IBoard CreateNewBoard(IBoard board, Movement movement, Color oldTurnColor)
+        {
+
+            List<Piece> newPieces = new List<Piece>();
```

Comment:
you can refactor this to something like we did with validators and generate tiles for piece